isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -869,266 +857,356 @@ namespace Isis {
+      fpOut << buf;
+      if (errorPropagation) {
+        sprintf(buf,"%*s",7,adjustedSigma.toStdString().c_str());
       }
-      // pointing parameters
-      for (int i = nPositionParameters; i < nParameters; i++) {
-        if (!useDefaultPointing) {
-          // If solving camera and not solving for twist, provide default values for twist to
-          // prevent bad indexing into m_corrections and m_*sigmas
-          // TWIST is last parameter, which corresponds to nParameters - nPointingCoefficients
-          if ( (i >= nParameters - nPointingCoefficients) && useDefaultTwist) {
-            correction = 0.0;
-            adjustedSigma = "N/A";
-            sigma = "N/A";
-          }
-          else {
-            correction = m_corrections(i - offset);
-            adjustedSigma = QString::number(m_adjustedSigmas(i-offset) * RAD2DEG, 'f', 8);
-            sigma = ( IsSpecial(m_aprioriSigmas[i - offset]) ? "FREE" :
-                    toString(m_aprioriSigmas[i-offset], 8) );
-          }
-        }
-        // We are using default pointing, so provide default correction and sigma values to output
-        else {
-          correction = 0.0;
-          adjustedSigma = "N/A";
-          sigma = "N/A";
-        }
-        if (errorPropagation) {
-          qStr = QString("%1%2%3%4%5%6\n").
-          arg( parameterNamesList.at(i) ).
-          arg( (finalParameterValues[i] - correction * RAD2DEG), 17, 'f', 8).
-          arg(correction * RAD2DEG, 21, 'f', 8).
-          arg(finalParameterValues[i], 20, 'f', 8).
-          arg(sigma, 18).
-          arg(adjustedSigma, 18);
-        }
-        else {
-          qStr = QString("%1%2%3%4%5%6\n").
-          arg( parameterNamesList.at(i) ).
-          arg( (finalParameterValues[i] - correction * RAD2DEG), 17, 'f', 8).
-          arg(correction * RAD2DEG, 21, 'f', 8).
-          arg(finalParameterValues[i], 20, 'f', 8).
-          arg(sigma, 18).
-          arg("N/A", 18);
-        }
-        finalqStr += qStr;
+      else {
+        sprintf(buf,"%s","N/A");
       }
+      fpOut<<buf;
+      sprintf(buf,"\t\t\t\t");
+      fpOut<<buf;
+      sprintf(buf,"%-*s\n",10,correctionUnitList.at(i).toStdString().c_str() );
+      fpOut<<buf;
 
     }
-    // this implies we're writing to images.csv
-    else {
-      // position parameters
-      for (int i = 0; i < nPositionParameters; i++) {
-        if (!useDefaultPosition) {
-          correction = m_corrections(i);
-          adjustedSigma = QString::number(m_adjustedSigmas[i], 'f', 8);
-          sigma = ( IsSpecial(m_aprioriSigmas[i]) ? "FREE" : toString(m_aprioriSigmas[i], 8) );
-        }
-        // Provide default values for position if not solving position
-        else {
+
+    // We need to use an offset of -3 (1 coef; X,Y,Z) if we used the default center coordinate
+    // (i.e. we did not solve for position), as m_corrections and m_*sigmas are populated
+    // according to which parameters are solved
+    int offset = 0;
```

Comment:
This offsetting is somewhat unnecessary and I think there is a cleaner way to address this. Mainly splitting the resulting Position and Pointing parameters into two different arrays in the fetchBundleOutputData call instead of a single massive array of parameters.